### PR TITLE
[API] Add error code for API responses

### LINKED ISSUE #6131: Add error code to API response outer field
Hello team,

With the changes to Wazuh API for 4.0, we removed error field from the json response and moved it inside the failed_items field, where every failed item would have its own error in case it occurred. However, keeping the outside error field for HTTP responses with code 200 will help the developer determine if everything went OK, everything failed or the response is only partially OK.

We can add this change in the API middleware in charge of processing the response. 

error field will come in every HTTP 200 response with the following values:

0 -> All petitions succeeded.
1 -> All petitions failed.
2 -> There is a partial response with some successes and some fails.

The HTTP response would look like:

```
                data:
                    affected_items:
                      - '002'
                      - '004'
                    total_affected_items: 2
                    failed_items: []
                    total_failed_items: 0
                error: 0
                message: 'Restart command was sent to all agents'
``

### PR DESCRIPTION
**Pull request by CarlosRS9**
Hello team,

This PR closes #6131. The api responses have been uptated to include an `error` field. This `error` replaces the old `code` field present in the responses with errors. Additionally, the HTTP `200` responses will contain the `error` field with the following values:

- `0`: A `200` response with only affected items
- `1`: A `200` response with only failed items
- `2`: A `200` response with failed and affected items.

### CODE REVIEW

File: api/api/controllers/agents_controller.py [hunk 1]
```diff
@@ -82,6 +82,7 @@ async def get_agents(request, pretty=False, wait_for_complete=False, list_agents
     :param ip: Filters by agent IP
     :return: AllItemsResponseAgents
     """
+
```
Comment (davidjiglesias) [inline api/api/controllers/agents_controller.py:85]:
```suggestion
```

File: api/api/util.py [hunk 2]
```diff
@@ -294,3 +294,26 @@ def raise_if_exc(obj):
         _create_problem(obj)
     else:
         return obj
+
+
+from aiohttp import web
```
Comment (davidjiglesias) [inline api/api/util.py:299]: Move import to top of the file and optimize imports

File: api/api/util.py [hunk 3]
```diff
@@ -294,3 +294,26 @@ def raise_if_exc(obj):
...
+
+def json_response_error_code(data=None, status=200, dumps=None, **kwargs):
```
Comment (davidjiglesias) [inline api/api/util.py:301]: Is this function finally used?

**Review by davidjiglesias — CHANGES_REQUESTED** [review]

File: api/api/util.py (hunk 3 above)
Comment (CarlosRS9) [reply api/api/util.py]: That function was deprecated. I have removed it

File: api/api/util.py (hunk 2 above)
Comment (CarlosRS9) [reply api/api/util.py]: The import is deprecated. I have removed it

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**